File: LeetCode/Easy/Problems 251-300/Problem 283/p283Sol1.c
void moveZeroes(int* nums, int numsSize){
    int i = 0; //Used to find zero elements
    int j = 1; //Used to find non-zero elements
    
    while (j < numsSize) {
        if (nums[i] == 0 && nums[j] == 0) {
            j++;
        }
        else if (nums[i] == 0 && nums[j] != 0) {
            int temp = nums[i];
            nums[i] = nums[j];
            nums[j] = temp;
            i++, j++;
        } 
        else {
            i++, j++;
        }
    }       
}

File: LeetCode/Easy/Problems 251-300/Problem 283/p283Sol1.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        i = 0 #Used to find zero elements
        j = 1 #Used to find non-zero elements 
        length = len(nums)
        
        while j < length:
            if nums[i] == 0 and nums[j] == 0:
                j += 1
            elif nums[i] == 0 and nums[j] != 0:
                nums[i], nums[j] = nums[j], nums[i]
                i += 1
                j += 1
            else:
                i += 1
                j += 1